assemblyline/odm/models/signature.py

```diff
@@ -11,18 +11,23 @@
 RULE_STATUSES = DEPLOYED_STATUSES + DRAFT_STATUSES + STALE_STATUSES
 
 
-@odm.model(index=True, store=True)
+@odm.model(index=True, store=True, description="""The Signature model within Assemblyline serves as a central framework for defining and managing security signatures, crucial components for the detection and analysis of malware. It provides a structured format that encompasses essential attributes such as a signature's name, classification level, source, and unique identifier. Moreover, it incorporates revision history, last modification timestamps, and statistical data to gauge performance and utility.
+
+The model presents users with a mix of static and dynamic information, ranging from immutable identification details to variable metadata reflecting the signature's current operational status and history of changes. This latter aspect is captured through fields that log the date and user associated with the last status update, offering a view into the signature's lifecycle.
+
+Understanding the Signature model is vital for cybersecurity professionals who are tasked with crafting precise Lucene-based search queries in Assemblyline. Mastery of this model's components will enable users to efficiently search, filter, and analyze signatures based on various parameters, thereby facilitating effective management and deployment in a cybersecurity context.
+""")
 class Signature(odm.Model):
-    classification = odm.Classification(store=True, default=Classification.UNRESTRICTED)
-    data = odm.Text(index=False, store=False)
-    last_modified = odm.Date(default="NOW")
-    name = odm.Keyword(copyto="__text__")
-    order = odm.Integer(default=1, store=False)
-    revision = odm.Keyword(default="1")
-    signature_id = odm.Optional(odm.Keyword())
-    source = odm.Keyword()
-    state_change_date = odm.Optional(odm.Date(store=False))
-    state_change_user = odm.Optional(odm.Keyword(store=False))
-    stats = odm.Compound(Statistics, default={})
-    status = odm.Enum(values=RULE_STATUSES, copyto="__text__")
-    type = odm.Keyword(copyto="__text__")
+    classification = odm.Classification(store=True, default=Classification.UNRESTRICTED, description="Indicates the sensitivity level of the signature, which dictates who can access it based on their clearance.")
+    data = odm.Text(index=False, store=False, description="Stores the actual signature data or pattern used for malware detection.")
+    last_modified = odm.Date(default="NOW", description="Records the timestamp of the most recent update to the signature. Defaults to the current time when the signature is modified.")
+    name = odm.Keyword(copyto="__text__", description="A unique and descriptive name for the signature.")
+    order = odm.Integer(default=1, store=False, description="**TODO**:Lower number means higher priority?  **Generated**:Determines the processing order of the signature relative to others. A lower number indicates higher priority.")
```

Comment:
@cccs-douglass I don't think this field really does anything, maybe a hold over from v3?

We should probably remove it if it's not being used